test/unit/kubernetes-deploy/options_helper_test.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'tempfile'
+
+class OptionsHelperTest < KubernetesDeploy::TestCase
+  def test_single_template_dir_only
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([fixture_path('hello-cloud')]) do |template_dir|
+      assert_equal(fixture_path('hello-cloud'), template_dir)
+    end
+  end
+
+  def test_multiple_template_dirs
+    template_dirs = [fixture_path('hello-cloud'), fixture_path('partials')]
+
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir(template_dirs) do |template_dir|
+      fixture_path_entries = template_dirs.collect { |dir| Dir.entries(dir) }.flatten.uniq
+      template_dir_entries = Dir.entries(template_dir)
+      assert_equal(fixture_path_entries.length, template_dir_entries.length)
+      fixture_path_entries.each do |fixture|
+        assert(template_dir_entries.select { |s| s.include?(fixture) })
```

Comment:
I think this will always pass, because `select` will return `[]` if it never matches, which is truthy. I suspect that's what's happening, because the two files will have different full paths, one being in a tempdir and one being in the original location.

Reply:
`refute(template_dir_entries.select { |template| template.include?(fixture) }.empty?)` passes the test, so the logic is working even though the test was initially flawed.

If you feel it's worth massaging this test some more I'll do so but I'm satisfied with it for now.